### PR TITLE
Move the dump parse tree logic slightly

### PR DESCRIPTION
**Pull request by shayne-fletcher-da**
Move dump code

### CODE REVIEW

File: compiler/damlc/lib/DA/Cli/Damlc.hs [hunk 1]
```diff
@@ -378,21 +378,22 @@ execCompile inputFile outputFile opts =
           setFilesOfInterest ide (Set.singleton inputFile)
           runAction ide $ do
             when (optWriteInterface opts') $ do
+
+                -- Support for '-ddump-parsed', '-ddump-parsed-ast', '-dsource-stats'.
```
Comment (hurryabit) [inline compiler/damlc/lib/DA/Cli/Damlc.hs]: Why is this conditional on writing interface files?

Comment (cocreature) [reply compiler/damlc/lib/DA/Cli/Damlc.hs]: Good catch, this should move outside and before the `when`.

Comment (shayne-fletcher-da) [reply compiler/damlc/lib/DA/Cli/Damlc.hs]: On it.

**Comment by shayne-fletcher-da** [comment]
@martin-drhu-da @cocreature i updated this inline with martin's observation that it still wasn't in quite the right place. want to have a quick look before i land it?

**Comment by shayne-fletcher-da** [comment]
Oops. Too late :)

**Pull request merged**